src/mon/FSCommands.cc

```diff
@@ -436,18 +437,38 @@ class SetHandler : public FileSystemCommandHandler
         return r;
       }
 
+      ss << fs->mds_map.get_fs_name();
+
       fsmap.modify_filesystem(
           fs->fscid,
           [is_down](std::shared_ptr<Filesystem> fs)
       {
-        if (is_down) {
-          fs->mds_map.set_flag(CEPH_MDSMAP_DOWN);
-        } else {
-          fs->mds_map.clear_flag(CEPH_MDSMAP_DOWN);
-        }
+	if (is_down) {
+	  fs->mds_map.set_flag(CEPH_MDSMAP_DOWN);
+	  fs->mds_map.set_old_max_mds();
+	  fs->mds_map.set_max_mds(0);
+	} else {
+	  mds_rank_t oldmax = fs->mds_map.get_old_max_mds();
+	  fs->mds_map.clear_flag(CEPH_MDSMAP_DOWN);
+	  fs->mds_map.set_max_mds(oldmax ? oldmax : 1);
+	}
       });
 
-      ss << "marked " << (is_down ? "down" : "up");
+      if (is_down) {
```

Comment:
It would be nice to make cluster_down kill all the active daemons (i.e. equivalent of "mds fail" on them), but I don't get the idea of shutting down the ranks completely.

As a user, if I want to switch my cluster off for some reason, then switch it back on again, I don't necessarily want to tear down all per-rank state, flush all journals, do a bunch of migration, etc etc.